interface/patient_file/summary/demographics.php

```diff
@@ -1223,12 +1224,29 @@ function setMyPatient() {
                     endif;
 
                     if ($GLOBALS['portal_onsite_two_enable']):
+                        /** @var EventDispatcher */
+                        $d = $GLOBALS['kernel']->getEventDispatcher();
+                        $e = new CardRenderEvent('portal');
+
+                        // Proof of concept, this gets moved before the PR gets merged
+                        $d->addListener(CardRenderEvent::EVENT_HANDLE, function($event) {
+                            if ($event->getCard() == 'portal') {
+                                $event->appendContent([
+                                    'template' => 'patient/partials/testing.html.twig',
+                                    'vars' => ['var1' => 'hello!'],
+                                ]);
+                            }
+                        });
+
+                        $result = $d->dispatch(CardRenderEvent::EVENT_HANDLE, $e);
+
                         echo $twig->getTwig()->render('patient/partials/portal.html.twig', [
                             'portalAuthorized' => portalAuthorized($pid),
                             'portalLoginHref' => $portal_login_href,
                             'title' => xl('Patient Portal'),
                             'id' => 'patient_portal',
                             'initiallyCollapsed' => (getUserSetting($id) == 0) ? false : true,
+                            'content' => $result->getContent(),
```

Comment:
Would these be better renamed as 'templates'? When I think of content I'm thinking of string / html content.

Reply:
Sure, content could be better. Maybe `data` instead? Because `getContent()` returns a template string and potentially variable data